src/git.js

```diff
@@ -218,6 +232,96 @@ class Git {
 		)
 	}
 
+	// Creates a tree object with all the blobs of each commit
+	async getTree(commitSha) {
+		const output = await execCmd(
+			`git ls-tree -r --full-tree ${ commitSha }`,
+			this.workingDir
+		)
+
+		const tree = []
+		for (const treeObject of output.split('\n')) {
+			const [ mode, type, sha ] = treeObject.split(/\s/)
+			const treeObjectPath = treeObject.split('\t')[1]
+
+			const treeEntry = {
+				mode,
+				type,
+				content: await this.getBlobContent(sha),
+				path: treeObjectPath
+			}
+			tree.push(treeEntry)
+		}
+		return tree
+	}
+
+	// Gets the commit list in chronological order
+	async getCommitsToPush() {
+		const output = await execCmd(
+			`git log --format=%H --reverse ${ this.baseBranch }..HEAD`,
+			this.workingDir
+		)
+
+		const commits = output.split('\n')
+		return commits
+	}
+
+	async getCommitMessage(commitSha) {
+		return await execCmd(
+			`git log -1 --format=%B ${ commitSha }`,
+			this.workingDir
+		)
+	}
+
+	// Returns an array of objects with the git tree and the commit, one entry for each pending commit to push
+	async getCommitsDataToPush() {
+		const commitsToPush = await this.getCommitsToPush()
+
+		const commitsData = []
+		for (const commitSha of commitsToPush) {
+			const commitData = {
+				commitMessage: await this.getCommitMessage(commitSha),
+				tree: await this.getTree(commitSha)
+			}
+			commitsData.push(commitData)
+		}
+		return commitsData
+	}
+
+	// A wrapper for running all the flow to generate all the pending commits using the GitHub API
+	async createGithubVerifiedCommits() {
+		const commitsData = await this.getCommitsDataToPush()
+
+		// Creates the PR branch if doesn't exists
+		try {
+			await this.github.git.createRef({
+				owner: this.repo.user,
+				repo: this.repo.name,
+				sha: this.lastCommitSha,
+				ref: 'refs/heads/' + this.prBranch
+			})
+
+			core.debug(`Created new branch ${ this.prBranch }`)
+		} catch (error) {
+			// If the branch exists ignores the error
+			if (error.message !== 'Reference already exists') throw error
+		}
+
+		for (const commitData of commitsData) {
+			await this.createGithubTreeAndCommit(commitData.tree, commitData.commitMessage)
+		}
+
+		core.debug(`Updating branch ${ this.prBranch } ref`)
+		await this.github.git.updateRef({
+			owner: this.repo.user,
+			repo: this.repo.name,
+			ref: `heads/${ this.prBranch }`,
+			sha: this.lastCommitSha,
+			force: true
+		})
+		core.debug(`Commit using GitHub API completed`)
+	}
+
```

Comment:
This function is a wrapper of the rest of the logic. 

Basically, it does: 
- Creates the branch if doesn't exist
- For each commit pending to push (to make compatible with the logic of `COMMIT_EACH_FILE` (calls the function createGithubTreeAndCommit)
  - Creates the tree on GitHub using the tree object.
  - Creates a commit with the git tree
- Updates the ref of the PR branch to the latest commit